### PR TITLE
Add functions to MSERDetector for detectRegions, Sync and Async

### PR DESCRIPTION
**Pull request by btsimonh**
Adds functions
MSERDetector.detectRegions( img )
MSERDetector.detectRegionsAsync( img, function( err, return ){ } );

Returns a structure { msers: [ array of Point2 arrays ], bboxes:[ array of bounding Rects ] } 
as per Opencv docs.

usage something like:

  var options = { delta:5, minArea:60, maxArea:14400,maxVariation:0.25,minDiversity:0.2,
	   maxEvolution:200,areaThreshold:1.01,minMargin:0.003,edgeBlurSize:5 };
  var det = new cv.MSERDetector( options );
  det.detectRegionsAsync(img, function (err, result) { 
   console.log(util.inspect(result));
  }

### CODE REVIEW

**Comment by justadudewhohacks** [comment]
That was fast. Great, thanks for adding :)

**Comment by btsimonh** [comment]
sorry I couldn't get my head around the testing scheme :(.

File: cc/modules/features2d/detectors/MSERDetector.cc [hunk 1]
```diff
@@ -57,4 +62,61 @@ NAN_METHOD(MSERDetector::New) {
...
+        for (int i = 0; i < mser_bbox.size(); i++) {
+            FF_OBJ jsR = FF_NEW_INSTANCE(Rect::constructor);
+            FF_UNWRAP_RECT_AND_GET(jsR) = mser_bbox.at(i);
+            jsBbox->Set(i, jsR);
+        }
+
+        FF_OBJ ret = FF_NEW_OBJ();            
+        Nan::Set(ret, FF_NEW_STRING("msers"), jsRegions);
+        Nan::Set(ret, FF_NEW_STRING("bboxes"), jsBbox);
+        return ret;
```
Comment (justadudewhohacks) [inline cc/modules/features2d/detectors/MSERDetector.cc]: This should be enough here:
``` c++
FF_OBJ ret = FF_NEW_OBJ();            
Nan::Set(ret, FF_NEW_STRING("msers"), ObjectArrayOfArraysConverter<Point2, cv::Point2d>::wrap(regions));
Nan::Set(ret, FF_NEW_STRING("bboxes"), ObjectArrayConverter<Rect, cv::Rect>::wrap(mser_bbox));
return ret;
```

That way you don't have to do all the conversion yourself. :)

**Comment by justadudewhohacks** [comment]
No worries, the testing is a little bit squishy. The intention is to simply make sure that values are passed to and from the OpenCV API correctly. I will eventually add some tests at some point.

**Comment by btsimonh** [comment]
updated as you highlighted :).
After extensive testing, I decided the possible memory leak was a red-herring.

**Comment by justadudewhohacks** [comment]
Your changes have been published with version 3.4.0.

**Comment by btsimonh** [comment]
thankyou! - now not sure about my USE of MSER, but at least it's now available :).

**Pull request merged**